File: backend/PriorAuthorization.Shared/Enums/RequestStatus.cs
namespace PriorAuthorization.Shared.Enums;

public enum RequestStatus : byte
{
    Draft = 1,
    VerificationFailed = 2,
    ReadyForSubmission = 3,

    Submitted = 4,
    UnderReview = 5,
    AdditionalInfoRequired = 6,
    ReSubmitted = 8,

    Approved = 7,
    Denied = 9,
    Expired = 10
}

File: frontend/src/utils/request-status.ts
import {
  RequestStatus
}
from "../enums/request-status.enum";

export const getStatusLabel =
  (
    status:
      RequestStatus
  ): string => {
    switch (status) {
      case RequestStatus.Draft:
        return "Draft";

      case RequestStatus.VerificationFailed:
        return "Verification Failed";

      case RequestStatus.ReadyForSubmission:
        return "Ready For Submission";

      case RequestStatus.Submitted:
        return "Submitted";

      case RequestStatus.UnderReview:
        return "Under Review";

      case RequestStatus.AdditionalInfoRequired:
        return "Additional Info Required";

      case RequestStatus.ReSubmitted:
        return "Re-Submitted";

      case RequestStatus.Approved:
        return "Approved";

      case RequestStatus.Denied:
        return "Denied";

      case RequestStatus.Expired:
        return "Expired";

      default:
        return "Unknown";
    }
  };